bundles/org.openhab.binding.synopanalyzer/src/main/java/org/openhab/binding/synopanalyzer/internal/stationdb/StationDbService.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.synopanalyzer.internal.stationdb;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link StationDbService} makes available a list of known Synop stations.
+ *
+ * @author Gaël L'hopital - Initial Contribution
+ */
+
+@Component(service = StationDbService.class)
+@NonNullByDefault
+public class StationDbService {
+    private final List<Station> stations;
+
+    @Activate
+    public StationDbService() {
+        try (InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream("/db/stations.json");
+                Reader reader = new InputStreamReader(is, StandardCharsets.UTF_8);) {
+            Gson gson = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).create();
+            stations = Arrays.asList(gson.fromJson(reader, Station[].class));
+        } catch (IOException e) {
```

Comment:
I am not sure that Gson exceptions are catched

Reply:
My request was not to remove this exception catch which has probably an utility but rather to mention that Gson exceptions are not caught. So I was more expecting additional exception catching.

Reply:
In case of any problem, you could log something and consider an empty list of stations.

Reply:
> My request was not to remove this exception catch which has probably an utility but rather to mention that Gson exceptions are not caught. So I was more expecting additional exception catching.

Sure, but Gson....create() does not raise any exception, so what should I be catching ?

Reply:
> In case of any problem, you could log something and consider an empty list of stations.

Yes, this is better.

Reply:
I am of course meaning the call to gson.fromJson which can trigger exceptions.
Of course, as you are parsing a local file you created in resources, the risk of exception is more or less null.
But that would be better to catch these exceptions.

Reply:
Sure. I don't know where I had the head .??.??..
Done.